src/mscorlib/shared/System/IO/PathInternal.Windows.StringBuffer.cs

```diff
@@ -27,14 +27,12 @@ internal static bool IsExtended(ref StringBuffer path)
         /// <summary>
         /// Gets the length of the root of the path (drive, share, etc.).
         /// </summary>
-        internal unsafe static int GetRootLength(ref StringBuffer path)
+        internal static int GetRootLength(ref StringBuffer path)
         {
-            if (path.Length == 0) return 0;
+            if (path.Length == 0)
+                return 0;
 
-            fixed (char* value = path.UnderlyingArray)
-            {
-                return GetRootLength(value, path.Length);
-            }
+            return GetRootLength(new ReadOnlySpan<char>(path.UnderlyingArray));
```

Comment:
Is this correct?  Shouldn't it be `new ReadOnlySpan<char>(path.UnderlyingArray, 0, path.Length)`?

Reply:
I guess the current version is also correct
```
        public ReadOnlySpan(T[] array)
        {
            if (array == null)
                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);

            _pointer = new ByReference<T>(ref Unsafe.As<byte, T>(ref array.GetRawSzArrayData()));
            _length = array.Length;
        }
```

Reply:
Note that length of the underlying array can differ from the length of the StringBuffer.